### PR TITLE
fix(create-changelog): update to new api

### PR DESCRIPTION
**Pull request by levithomason**
This PR updates our circle changelog script to work with the new [github changelog api](https://github.com/levithomason/github-changelog-api).  Usage remains the same:

```sh
USAGE: ta-script circle_ci/create_changelog [options]

Options:
  -t, --token  github personal access token
  --help       Show help                                               [boolean]
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+## [v2.4.2](https://github.com/TechnologyAdvice/ta-scripts/tree/v2.4.2) (2016-07-04)
+[Full Changelog](https://github.com/TechnologyAdvice/ta-scripts/compare/v2.4.1...v2.4.2)
+
+## [v2.4.1](https://github.com/TechnologyAdvice/ta-scripts/tree/v2.4.1) (2016-06-27)
+[Full Changelog](https://github.com/TechnologyAdvice/ta-scripts/compare/v2.4.0...v2.4.1)
+
+**Merged pull requests:**
+
+- Add git checks to npm release [\#24](https://github.com/TechnologyAdvice/ta-scripts/pull/24) ([levithomason](https://github.com/levithomason))
+
```
Comment (levithomason) [inline CHANGELOG.md]: Updated our log while at it, for testing.

File: circle_ci/create_changelog.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+const axios = require('axios')
+const childProcess = require('child_process')
+const fs = require('fs-promise')
+
+// ----------------------------------------
+// Environment
+// ----------------------------------------
+
+const { CIRCLE_BRANCH, CIRCLE_PROJECT_USERNAME, CIRCLE_PROJECT_REPONAME } = process.env
+if (CIRCLE_BRANCH) throw new Error('Missing env var CIRCLE_BRANCH.')
```
Comment (davezuko) [inline circle_ci/create_changelog.js]: Should these have a `!`?

File: circle_ci/create_changelog.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      },
+      err => {
+        if (err.response.status === 302) {
+          // poll
+          log('Changelog job in progress, waiting...', formatRes(err.response))
+          return wait(5000).then(pollForChangelog)
+        } else {
+          // reject all other errors
+          log('Error while polling', formatRes(err.response))
+          process.exit(1)
```
Comment (davezuko) [inline circle_ci/create_changelog.js]: I don't think this should necessarily exit, but rather have the final promise chain that runs all of these be responsible for exiting.

File: circle_ci/create_changelog.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    })
+}
+
+const commitAndPush = () => new Promise((resolve, reject) => {
+  log('Committing and pushing changelog, if needed')
+  try {
+    const changelogWasUpdated = childProcess.execSync('git ls-files -m')
+      .toString()
+      .split('\n')
+      .filter(l => !!l)
```
Comment (davezuko) [inline circle_ci/create_changelog.js]: Totally fine, but cool trick I learned from the React team FWIW:

`.filter(Boolean)`

File: circle_ci/create_changelog.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+const writeChangelog = (data) => {
+  log('Writing changelog to disk:', data)
+  return fs.writeFile('CHANGELOG.md', data)
+    .catch(err => {
+      log('Failed to write changelog to disk:', formatRes(err.response))
+      process.exit(1)
+    })
+}
+
+const commitAndPush = () => new Promise((resolve, reject) => {
```
Comment (davezuko) [inline circle_ci/create_changelog.js]: Can get rid of your try/catch with...

```js
const commitAndPush = (data) => Promise.resolve()
  .then(() => {
    // if anything throws in here, the promise will be rejected.
  })
```

Comment (davezuko) [inline circle_ci/create_changelog.js]: Also lets you get rid of the meaningless `resolve()` call.

File: circle_ci/create_changelog.js (hunk 3 above)
Comment (davezuko) [inline circle_ci/create_changelog.js]: Note: this applies to all functions.

File: circle_ci/create_changelog.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+const axios = require('axios')
+const childProcess = require('child_process')
+const fs = require('fs-promise')
+
+// ----------------------------------------
+// Environment
+// ----------------------------------------
+
+const { CIRCLE_BRANCH, CIRCLE_PROJECT_USERNAME, CIRCLE_PROJECT_REPONAME, CIRCLE_USERNAME } = process.env
+if (CIRCLE_BRANCH)            throw new Error('Missing env var CIRCLE_BRANCH.')
```
Comment (davezuko) [inline circle_ci/create_changelog.js]: Shouldn't this be `if (!CIRCLE_BRANCH)` (missing `!`)?

Comment (levithomason) [inline circle_ci/create_changelog.js]: Fixed all

File: circle_ci/create_changelog.js (hunk 5 above)
Comment (levithomason) [inline circle_ci/create_changelog.js]: Hm, I don't think it should _always_ resolve as it then can't be moved around and relied on to resolve on success / reject on error.

File: circle_ci/create_changelog.js (hunk 3 above)
Comment (levithomason) [inline circle_ci/create_changelog.js]: I wrestled with this for a few minutes, a final `.catch()` always  exited 0, no matter what I did (threw, process.exit, etc).  Will give it one more go with the refactors made.

File: circle_ci/create_changelog.js (hunk 5 above)
Comment (davezuko) [inline circle_ci/create_changelog.js]: ?? Maybe misunderstanding, but this example does the same exact thing as your code but gets rid of the need for you to wrap your code in a try/catch and manually resolve a promise.

**Pull request merged**